app/policies/transaction_policy.rb

```diff
@@ -1,13 +1,22 @@
 class TransactionPolicy < ApplicationPolicy
   def create?
-    Pundit.policy!(user, record.parent_activity).create?
+    Pundit.policy!(user, record.parent_activity).create? && !!associated_submission&.active?
   end
 
   def update?
-    Pundit.policy!(user, record.parent_activity).update?
+    Pundit.policy!(user, record.parent_activity).update? && !!associated_submission&.active?
   end
 
   def destroy?
     Pundit.policy!(user, record.parent_activity).destroy?
   end
+
+  private
+
+  def associated_submission
```

Comment:
I'm late to reading through this because of other things on my plate -- sorry! I just wanted to flag this logic as something we might want to refactor out as it duplicates some functionality in `CreateTransaction` below, possibly we can have a look at this while doing bulk upload. But even if we don't, just using this policy and `CreateTransaction` to implement bulk upload should make its behaviour consistent :)